src/draw.c

```diff
@@ -742,26 +742,48 @@ static void render_content(cairo_t *c, struct colored_layout *cl, int width, dou
 
                 // draw progress bar
                 // Note: the bar could be drawn a bit smaller, because the frame is drawn on top
-                // left side
+                // left side (fill)
                 cairo_set_source_rgba(c, cl->highlight.r, cl->highlight.g, cl->highlight.b, cl->highlight.a);
-                draw_rect(c, x_bar_1, frame_y, progress_width_1, progress_height, scale);
+                if(settings.progress_bar_corner_radius > 0) {
```

Comment:
I think the checks for the corner radius are unnecesary. In the other places we use draw_rounded_rect we don't check for that either. For optmization purposes, the check could be added in `draw_rounded_rect` itself if we want.
So I'd suggest removing the check here at least.